engine/access/rest/handlers.go

```diff
@@ -0,0 +1,228 @@
+package rest
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/gorilla/mux"
+	"github.com/rs/zerolog"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/onflow/flow-go/access"
+	"github.com/onflow/flow-go/engine/access/rest/generated"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+const BlockIDCntLimit = 50
+
+var MaxAllowedBlockIDsCnt = BlockIDCntLimit
+
+// Handlers provide collection of handlers used by the API server
+type Handlers struct {
+	backend access.API
+	logger  zerolog.Logger
+}
+
+func NewHandlers(backend access.API, logger zerolog.Logger) *Handlers {
+	return &Handlers{
+		backend: backend,
+		logger:  logger,
+	}
+}
+
+func (h *Handlers) BlocksIdGet(w http.ResponseWriter, r *http.Request) {
+	// create h logger for the request
+	errorLogger := h.logger.With().Str("request_url", r.URL.String()).Logger()
+
+	w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+
+	vars := mux.Vars(r)
+	idParam := vars["id"]
+
+	// gorilla mux retains opening and ending square brackets for ids
+	idParam = strings.TrimSuffix(idParam, "]")
+	idParam = strings.TrimPrefix(idParam, "[")
```

Comment:
Why are there opening and ending square brackets in the first place? They should not have been present in the query string...

Take a look at the table here: https://swagger.io/docs/specification/serialization/

The request should just look something like "v1/blocks/id1,id2,id3"

But maybe I'm misunderstanding how this works.

Reply:
this is because of how the Generated Go Client is sending the request. It does a `fmt.Sprintf("%v", id)` [here](https://github.com/onflow/flow/blob/master/openapi/go-client-generated/api_blocks.go#L196) which causes the ids to be written out as `[id1 id2 id3]` etc. given how `%v` [works](https://play.golang.org/p/NMNcaMeUdNM) for `[]string`.

Alternatively, I could write my own client or use a different tool to generate the client but to make progress on the implementation I added line 47 and 48  to just remove the extra chars from the request if present (if not then its no-op)

Reply:
makes sense and I think the workaround is ok for now. can you just add a comment explaining that to the code so we remember later.

if you haven't already, can you create an issue to track fixing the client?

Reply:
https://github.com/onflow/flow/issues/698 - on the flow repo to fix the client (assigned to me)

Reply:
and added comment to code explaining the same.